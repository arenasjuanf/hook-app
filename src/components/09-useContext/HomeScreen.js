import React from 'react'

export const HomeScreen = () => {
    return (
        <div>
            <h1>HomesCreen</h1>
            <hr/>
        </div>
    )
}
