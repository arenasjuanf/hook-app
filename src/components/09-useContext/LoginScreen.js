import React, {useContext} from 'react'
import { UserContext } from './UserContext'


export const LoginScreen = () => {

    const {setUser} = useContext(UserContext);

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr/>
            <button type="button" className="btn btn-primary" onClick={ () => setUser({id:123, name: "juanf"})}>
                Login
            </button>
        </div>
    )
}
