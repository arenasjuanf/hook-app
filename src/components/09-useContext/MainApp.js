import React from 'react'
import { AppRouter } from './AppRouter'

export const MainApp = () => {
    return (
      
        /*  <h1>MainApp</h1>
        <hr/> */
        <AppRouter />
        
    )
}
