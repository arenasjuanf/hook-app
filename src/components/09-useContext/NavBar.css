ul{
    list-style: none;
    display: flex;
    margin: 0px;
    padding: 2px;
}

li{
    margin: 5px;
}