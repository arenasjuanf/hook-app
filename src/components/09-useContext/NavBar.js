import React from 'react';
import {
    NavLink,
} from "react-router-dom";

import './NavBar.css';

export const NavBar = () => {
    return (
       
        <nav className="navbar navbar-dark bg-dark navbar-expand-lg">
            <div className="container-fluid">
                <a className="navbar-brand" href="#">UseContext</a>
                <button className="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
                    <span className="navbar-toggler-icon"></span>
                </button>
                <div className="collapse navbar-collapse" id="navbarNav">
                <ul className="navbar-nav">
                    
                    <NavLink exact activeClassName="active" className="nav-item nav-link" to="./">Home</NavLink> 
                
                
                    <NavLink exact activeClassName="active" className="nav-item nav-link" to="./about">About</NavLink> 
                
                
                    <NavLink exact activeClassName="active" className="nav-item nav-link" to="./login">Login</NavLink> 
                    
                   
                </ul>
                </div>
            </div>
        </nav>
    )
}
