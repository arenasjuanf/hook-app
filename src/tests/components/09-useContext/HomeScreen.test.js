import { shallow } from "enzyme";
import { HomeScreen } from "../../../components/09-useContext/HomeScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe("<HomeScreen/> tests", () =>{
    const wrapper = shallow(
        <UserContext.Provider value={{user: "juanf", email:"nada@nothing.com"}}>
            <HomeScreen/>
        </UserContext.Provider>
    );

    test('it should show successfully', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
})