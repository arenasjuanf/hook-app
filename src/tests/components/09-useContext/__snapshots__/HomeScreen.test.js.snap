// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen/> tests it should show successfully 1`] = `<HomeScreen />`;
